### ISSUE TITLE
Feature Request: Vec3,vec4 Color Picker

### ISSUE THREAD

**Issue by willstall** [actarian/vscode-glsl-canvas#13]
I use this plugin every single day for my Glsl daily and just love it. The one thing I'm really missing is the color picker from glslCanvas.....I also love the float sliders, but hey....I just love the color picker more.

Would there be any chance of integrating a color picker?

**Comment by leocb** [comment]
We all want this feature, believe me!
Currently VSCode does not support inline HTML/JS so we can't do this.. yet
There are some discussion about this (Microsoft/vscode#3220 and Microsoft/vscode#66493) and a PR that may be what we need (Microsoft/vscode#66418)

**Comment by actarian** [comment]
@willstall @leocb this functionality has been implemented and will be shipped with the next release of the extension. it essentially use DocumentColorProvider from the vscode Api

**Comment by willstall** [comment]
yooooooooooooooooooooo.....that's amazing news!!!

**Comment by leocb** [comment]
Nice! this will be a good workaround (tough not perfect because thebookofshaders's pickers are awesome) I guess this will do while we wait for the vscode features I linked to. I think we should still leave this issue opened(?)

**Comment by actarian** [comment]
@willstall @leocb feature added to last release. let's keep it opened waiting for code inset feature.

**Comment by willstall** [comment]
Got it working. Gonna give this a week or two and drop feedback in here =)

**Comment by willstall** [comment]
I'm loving this so far....

**Comment by willstall** [comment]
https://code.visualstudio.com/api/extension-guides/webview

Could this help for making a float component?

**Comment by leocb** [comment]
The feature you linked only displays webviews on a separate tab (some examples are: this GL canvas extension and the markdown preview), I looked into using it, but the user experience is just terrible and slow, the pickers need to be placed in-line or over the code to be useful in any way.

What we need is the code inset feature, that is the vscode#66418 PR I mentioned above, and that is still under development.

**Comment by willstall** [comment]
Ahhh yeah.... Figured as much. I got there following that link and threw it in real quick. I does look like some progress on that inset feature has been made in the last few weeks. I saw something in my searching about it

**Comment by willstall** [comment]
![Screen Shot 2019-07-11 at 12 53 58 PM](https://user-images.githubusercontent.com/8049767/61069901-57efa800-a3db-11e9-8f27-3614e3882b78.png)

Hey guys! Does this help at all? Looks like inset Webview's at a line number

**Comment by leocb** [comment]
Yes, that's the inset feature we're waiting for, it's in beta test on the Insiders build of vscode

**Comment by actarian** [comment]
🎉 Huray!

**Comment by err123** [comment]
Wow i love this

**Comment by kei-arc** [comment]
Is there a way to turn off this feature? 
In my case, this harms the readability a lot.
